scrapy/commands/startproject.py

```diff
@@ -43,41 +44,68 @@ def _module_exists(module_name):
         if not re.search(r'^[_a-zA-Z]\w*$', project_name):
             print('Error: Project names must begin with a letter and contain'\
                     ' only\nletters, numbers and underscores')
-        elif exists(project_name):
-            print('Error: Directory %r already exists' % project_name)
         elif _module_exists(project_name):
             print('Error: Module %r already exists' % project_name)
         else:
             return True
         return False
 
+    def _copytree(self, src, dst):
+        ignore = IGNORE
+        names = os.listdir(src)
+        ignored_names = ignore(src, names)
+
+        if not os.path.exists(dst):
+            os.makedirs(dst)
+
+        for name in names:
+            if name in ignored_names:
+                continue
+
+            srcname = os.path.join(src, name)
+            dstname = os.path.join(dst, name)
+            if os.path.isdir(srcname):
+                self._copytree(srcname, dstname)
+            else:
+                copy2(srcname, dstname)
+        copystat(src, dst)
+
     def run(self, args, opts):
-        if len(args) != 1:
+        if len(args) not in (1, 2):
             raise UsageError()
+
         project_name = args[0]
+        project_dir = args[0]
+
+        if len(args) == 2:
+            project_dir = args[1]
+            if exists(join(project_dir, 'scrapy.cfg')):
```

Comment:
Since we're using a custom copytree that now works in existing directories, I think we shouldn't assume the directory _will_ be created.
Users can now do `mkdir myproj; scrapy startproject myproj` and scrapy will work without complaining, therefore the extra check is a good idea.

In fact, I just tried running `scrapy startproject myproj` twice and it breaks the second time.